Testing/functional/tests/cypress/integration/e2e/dependent/report.js

```diff
@@ -0,0 +1,446 @@
+                ],
+            },
+            datasets: [
+                {
+                    name: "clinicalDocument",
+                    enabled: true,
+                },
+                {
+                    name: "covid19TestResult",
+                    enabled: true,
+                },
+                {
+                    name: "healthVisit",
+                    enabled: true,
+                },
+                {
+                    name: "hospitalVisit",
+                    enabled: true,
+                },
+                {
+                    name: "immunization",
+                    enabled: true,
+                },
+                {
+                    name: "labResult",
+                    enabled: true,
+                },
+                {
+                    name: "medication",
+                    enabled: true,
+                },
+                {
+                    name: "note",
+                    enabled: true,
+                },
+                {
+                    name: "specialAuthorityRequest",
+                    enabled: true,
+                },
+            ],
+        });
+        cy.login(
+            Cypress.env("keycloak.username"),
+            Cypress.env("keycloak.password"),
+            AuthMethod.KeyCloak,
+            "/dependents"
+        );
+    });
+
+    it("Validate Service Selection", () => {
+        const hdid = existingDependent.hdid;
+
+        const cardSelector = getCardSelector(hdid);
+        const tabButtonSelector = getTabButtonSelector(hdid, "report");
+        const tabSelector = `${cardSelector} [data-testid=report-tab]`;
+
+        cy.get(tabButtonSelector)
+            .should("be.visible")
+            .should("not.be.disabled")
+            .should("not.have.class", "disabled")
+            .click();
+
+        cy.get(`${tabSelector} [data-testid=exportRecordBtn] button`).should(
+            "be.disabled",
+            "be.visible"
+        );
+
+        cy.get(`${tabSelector} [data-testid=infoText]`).should(
+            "have.text",
+            " Select a record type above to create a report "
+        );
+
+        // display visual when no record type selected (mobile and desktop)
+        cy.get(`${tabSelector} [data-testid=infoImage]`).should("be.visible");
+        cy.viewport("iphone-6");
+        cy.get(`${tabSelector} [data-testid=infoImage]`).should("be.visible");
+        cy.viewport(1440, 600);
+
+        cy.get(`${tabSelector} [data-testid=reportType]`).select(
+            "COVID‑19 Tests"
+        );
+
+        cy.get(`${tabSelector} [data-testid=exportRecordBtn] button`).should(
+            "be.enabled",
+            "be.visible"
+        );
+
+        cy.get(`${tabSelector} [data-testid=reportType]`).select("");
+
+        cy.get(`${tabSelector} [data-testid=exportRecordBtn] button`).should(
+            "be.disabled",
+            "be.visible"
+        );
+    });
+
+    it.skip("Validate Medication Report", () => {
```

Comment:
Much of the individual tests could be standardised with the a method to select the report type etc.

Could we not just be testing the `[data-testId=reportSample]` to see if the preview is displayed? do we need to test each specific implementation of the report previews?

If the preview section can also be standardised we can reduce the maintenance overhead and simplify the entire structure into a method that test each option from start to finish?

Reply:
I think this is a good conversation to have with the dev team. For now I have followed our standard test approach.